semgrep-core/src/parsing/tree_sitter/Parse_swift_tree_sitter.ml

```diff
@@ -2702,17 +2952,26 @@ and map_type_constraint (env : env) (x : CST.type_constraint) =
       let v2 = map_identifier env v2 in
       let v3 = (* ":" *) token env v3 in
       let v4 = map_possibly_implicitly_unwrapped_type env v4 in
-      todo env (v1, v2, v3, v4)
+      let base_type = G.ArgType (G.TyN (H2.name_of_ids v2) |> G.t) in
+      let conformed_protocol = G.ArgType v4 in
+      G.NamedAttr
+        ( v3,
+          H2.name_of_id ("ConformsTo", v3),
```

Comment:
Probably better to use the string in the original program text via `str` even though it's just a colon.